test/parallel/test-fs-mkdtemp.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+common.refreshTmpDir();
+
+fs.mkdtempSync(common.tmpDir + '/foo.XXXXXX');
+assert(fs.readdirSync(common.tmpDir).some(function(folder) {
+    return folder.indexOf('foo.') === 0 && folder.length === 'foo.XXXXXX'.length;
+}));
+
+fs.mkdtemp(common.tmpDir + '/bar.XXXXXX', function(err, folder) {
+    assert(common.fileExists(folder));
+});
+
+fs.mkdtemp(common.tmpDir + '/bar', function(err) {
+    assert(err !== null);
```

Comment:
The return code should be EINVAL in that case. (The OSX man page doesn't say anything about EINVAL, but we can look in the code if need be).